dali/pipeline/operators/geometric/bb_flip_test.cc

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <tuple>
+#include "dali/test/dali_test_single_op.h"
+
+namespace dali {
+
+namespace {
+
+constexpr int kBbStructSize = 4;
+
+/**
+ * Roi can be represented in two ways:
+ * 1. Upper-left corner, width, height
+ *    (x1, y1,  w,  h)
+ * 2. Upper-left and Lower-right corners
+ *    (x1, y1, x2, y2)
+ *
+ * Both of them have coordinates in image coordinate system
+ * (i.e. 0.0-1.0)
+ */
+using Roi = std::array<float, kBbStructSize>;
+
+constexpr int kTestDataSize = 10;
+
+/**
+ * Test data for BbFlip operator. Data consists of:
+ * 0 -> reference data input
+ * 1 -> reference data horizontally flipped
+ * 2 -> reference data vertically flipped
+ */
+using TestData = std::array<std::tuple<Roi, Roi, Roi>, kTestDataSize>;
+
+// XXX: Looks hideous, but that's the only way in C++11
```

Comment:
You could always:
using TestCase = std::tuple<Roi, Roi, Roi>;
And the problem is not with C++11 but the particular libstdc++/compiler that we use.